File: src/wms/model/client/Recipient.java
package wms.model.client;

import javax.persistence.DiscriminatorValue;
import javax.persistence.Entity;
import javax.persistence.Table;

@Entity
@Table(name = "client")
@DiscriminatorValue(value = "2")
public class Recipient extends Client {
	private static final long serialVersionUID = -6775441124171801207L;

	public Recipient() {
		super();
	}

}

File: WebContent/page/app/model/entity/Recipient.js
/*
 * Project: MajekWMS
 * User   : kornicameister
 * Date   : 04.11.12
 * Time   : 01:26
 */

Ext.define('WMS.model.entity.Recipient', {
    extend: 'WMS.model.entity.Client',
    proxy : {
        type       : 'wms',
        url        : 'wms/agent/client',
        extraParams: {
            type: 'recipient'
        },
        writer     : {
            type          : 'json',
            root          : 'data',
            allowSingle   : false,
            writeAllFields: false,
            getRecordData : function (record, operation) {
                return WMS.model.entity.Client.getRecordData(record, operation);
            }
        }
    }
});